### ISSUE TITLE
Support for signedVersion attribute in azurerm_storage_account_sas

### ISSUE THREAD

**Issue by sschmeck** [terraform-providers/terraform-provider-azurerm#7619]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Currently the `signedVersion` of `ComputeAccountSASToken` is fix and set to `2017-07-29`. That version is outdated at least for the ADLS Gen2 access. It results in `400 - The specified Rest Version is Unsupported.`

A manually created account SAS token in Azure Portal sets `sv=2019-10-10`. This token works properly with ADLS Gen2.

Therefore I suggest to provide an optional argument `signed_version` for  `azurerm_storage_account_sas` data source.

### Affected Resource(s)

* azurerm_storage_account_sas

### Potential Terraform Configuration

```hcl
data "azurerm_storage_account_sas" "example" {
  connection_string = azurerm_storage_account.example.primary_connection_string
  https_only        = true
  
  // new argument
  signed_version = "2019-10-10"

  // ...
}
```

### References

* #6519 requests the same argument for `azurerm_storage_account_blob_container_sas` data source.

**Comment by jaicao** [comment]
We're using the azurerm_virtual_machine_extension resource on a windows virtual machine to enable the IaaSDiagnostics
```hcl
resource "azurerm_virtual_machine_extension" "diagnostics_agent" {

  name                       = var.name
  virtual_machine_id         = var.virtual_machine_id
  publisher                  = "Microsoft.Azure.Diagnostics"
  type                       = "IaaSDiagnostics"
  type_handler_version       = "1.9"
  auto_upgrade_minor_version = true

  settings = ....

  protected_settings = jsonencode({
    storageAccountName     = var.storage_account_name
    storageAccountSasToken = trimprefix(data.azurerm_storage_account_sas.example.sas), "?")
  })
```
When using the sas token generated by the data.azurerm_storage_account_sas.example.sas, we get this:
Error: Code="VMExtensionProvisioningError" Message="VM has reported a failure when processing extension '**name**'. Error message: \"No primary storage account found! See the troubleshooting guide to diagnose the problem: https://go.microsoft.com/fwlink/?linkid=852271\"\r\n\r\nMore information on troubleshooting is available at https://aka.ms/vmextensionwindowstroubleshoot "

However, when using a sas token generated by az cli, this resource suceeds.
command: _az storage account generate-sas --account-name <example_sa_name> --expiry 2037-12-31T23:59:00Z --permissions wlacu --resource-types co --services bt -o tsv_

It seems that the signed version is 2018-03-28 from the az cli.

**Comment by sschmeck** [comment]
Regarding the ADLS Gen2, the problem is not the authorization API version but the ADSL Gen2 API version. It's derived from the SAS token when not set explicitly. Therefore you can set a header with `x-ms-version: 2019-10-10`, which works with generated SAS token.

**Comment by tombuildsstuff** [comment]
Best I can tell this has been fixed in #8020 - closing
